LSP crashes?

```
����: Internal error: java.lang.IllegalArgumentException: object is not an instance of declaring class
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: object is not an instance of declaring class
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: object is not an instance of declaring class
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.checkReceiver(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.aya.prover.merged.module@0.20-SNAPSHOT/org.eclipse.lsp4j.jsonrpc.services.EndpointProxy.invoke(Unknown Source)
	at jdk.proxy2/jdk.proxy2.$Proxy10.clearAyaProblems(Unknown Source)
	at aya.lsp/org.aya.lsp.server.AyaService.clearProblems(Unknown Source)
	at aya.lsp/org.aya.lsp.server.AyaService$CallbackAdvisor.notifyIncrementalJob(Unknown Source)
	at aya.cli/org.aya.cli.library.LibraryCompiler.make(Unknown Source)
	at aya.cli/org.aya.cli.library.LibraryCompiler.make(Unknown Source)
	at aya.cli/org.aya.cli.library.LibraryCompiler.make(Unknown Source)
	at aya.cli/org.aya.cli.utils.AyaCompiler.catching(Unknown Source)
	at aya.cli/org.aya.cli.library.LibraryCompiler.start(Unknown Source)
	at aya.lsp/org.aya.lsp.server.AyaService.loadLibrary(Unknown Source)
	at kala.collection/kala.collection.internal.view.SeqViews$FlatMapped.lambda$iterator$0(Unknown Source)
	at kala.base/kala.collection.base.Iterators$3.next(Unknown Source)
	at kala.base/kala.collection.base.Iterators$ConcatAll.hasNext(Unknown Source)
	at kala.collection/kala.collection.immutable.ImmutableVector.from(Unknown Source)
	at kala.collection/kala.collection.immutable.ImmutableVector.from(Unknown Source)
```

@imkiva save me save me

```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.UnsupportedOperationException
	at org.eclipse.lsp4j.services.LanguageServer.setTrace(LanguageServer.java:115)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 13 more
```

--- a/lsp/src/main/java/org/aya/lsp/server/AyaLanguageClient.java
+++ b/lsp/src/main/java/org/aya/lsp/server/AyaLanguageClient.java
@@ -21,29 +21,31 @@ import java.util.Collections;
 import java.util.stream.Collectors;
 
 public interface AyaLanguageClient extends LanguageClient {
-  default void publishAyaProblems(
+  static void publishAyaProblems(
+    @NotNull AyaLanguageClient self,
     @NotNull ImmutableMap<Path, ImmutableSeq<Problem>> problems,
     @NotNull DistillerOptions options
   ) {
     problems.forEach((filePath, value) -> {
-      Log.d("Found %d issues in %s", value.size(), filePath);
+      Log.i("Found %d issues in %s", value.size(), filePath);
       var param = new PublishDiagnosticsParams(
         filePath.toUri().toString(),
         value
           .collect(Collectors.groupingBy(Problem::sourcePos, ImmutableSeq.factory()))
           .entrySet().stream()
           .map(kv -> toDiagnostic(kv.getKey(), kv.getValue(), options))
-          .collect(Collectors.toList())
+          .toList()
       );
-      publishDiagnostics(param);
+      self.publishDiagnostics(param);
     });
   }
 
-  default void clearAyaProblems(@NotNull ImmutableSeq<Path> files) {
-    files.forEach(f -> {
-      var param = new PublishDiagnosticsParams(f.toUri().toString(), Collections.emptyList());
-      publishDiagnostics(param);
-    });
+  static void clearAyaProblems(
+    @NotNull AyaLanguageClient self,
+    @NotNull ImmutableSeq<Path> files
+  ) {
+    files.forEach(f -> self.publishDiagnostics(
+      new PublishDiagnosticsParams(f.toUri().toString(), Collections.emptyList())));
   }
 
   private static @NotNull Diagnostic toDiagnostic(@NotNull SourcePos sourcePos, @NotNull Seq<Problem> problems, @NotNull DistillerOptions options) {

--- a/lsp/src/main/java/org/aya/lsp/server/AyaService.java
+++ b/lsp/src/main/java/org/aya/lsp/server/AyaService.java
@@ -171,22 +171,13 @@ public class AyaService implements WorkspaceService, TextDocumentService {
         Collectors.mapping(t -> t._2, ImmutableSeq.factory())
       ));
     var from = ImmutableMap.from(diags);
-    try {
-      client.publishAyaProblems(from, options);
-    } catch (Exception e) {
-      System.out.println(client.getClass());
-      System.out.println(from.getClass());
-    }
+    AyaLanguageClient.publishAyaProblems(client, from, options);
   }
 
   private void clearProblems(@NotNull ImmutableSeq<ImmutableSeq<LibrarySource>> affected) {
     if (client == null) return;
     var files = affected.flatMap(i -> i.map(LibrarySource::file));
-    try {
-      client.clearAyaProblems(files);
-    } catch (Exception e) {
-      System.out.println(client.getClass());
-    }
+    AyaLanguageClient.clearAyaProblems(client, files);
   }
 
   @Override public void didChangeWatchedFiles(@NotNull DidChangeWatchedFilesParams params) {

--- a/lsp/src/main/java/org/aya/lsp/utils/Resolver.java
+++ b/lsp/src/main/java/org/aya/lsp/utils/Resolver.java
@@ -1,4 +1,4 @@
-// Copyright (c) 2020-2022 Yinsen (Tesla) Zhang.
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
 // Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
 package org.aya.lsp.utils;
 
@@ -26,6 +26,8 @@ import org.aya.util.error.SourcePos;
 import org.aya.util.error.WithPos;
 import org.eclipse.lsp4j.Position;
 import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
 
 public interface Resolver {
   /** resolve a symbol by its qualified name in the whole library */
@@ -198,9 +200,7 @@ public interface Resolver {
       // for imported serialized definitions, let's compare by qualified name
       return var instanceof DefVar<?, ?> defVar
         && check instanceof DefVar<?, ?> checkDef
-        && defVar.module != null
-        && checkDef.module != null
-        && defVar.module.equals(checkDef.module)
+        && Objects.equals(defVar.module, checkDef.module)
         && defVar.name().equals(checkDef.name());
     }
   }

--- a/lsp/src/main/java/org/aya/lsp/utils/XY.java
+++ b/lsp/src/main/java/org/aya/lsp/utils/XY.java
@@ -1,4 +1,4 @@
-// Copyright (c) 2020-2022 Yinsen (Tesla) Zhang.
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
 // Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
 package org.aya.lsp.utils;
 
@@ -13,6 +13,6 @@ public record XY(int x, int y) {
   }
 
   public boolean inside(@NotNull SourcePos sourcePos) {
-    return sourcePos.contains(x, y);
+    return sourcePos.containsVisually(x, y);
   }
 }

--- a/tools/src/main/java/org/aya/util/error/SourcePos.java
+++ b/tools/src/main/java/org/aya/util/error/SourcePos.java
@@ -1,4 +1,4 @@
-// Copyright (c) 2020-2022 Yinsen (Tesla) Zhang.
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
 // Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
 package org.aya.util.error;
 
@@ -99,6 +99,10 @@ public record SourcePos(
 
   public boolean contains(int line, int column) {
     return line >= startLine && line <= endLine && column >= startColumn && column <= endColumn;
+  }
+
+  public boolean containsVisually(int line, int column) {
+    return line >= startLine && line <= endLine && column >= startColumn - 1 && column <= endColumn;
   }
 
   public boolean contains(int pos) {

